db/compaction_picker.cc

```diff
@@ -1373,6 +1356,39 @@ bool LevelCompactionBuilder::PickFileToCompact() {
   return start_level_inputs_.size() > 0;
 }
 
+bool LevelCompactionBuilder::PickL0ToBaseCompaction() {
```

Comment:
Although if there's a hotspot it typically won't be the oldest file. It also typically won't be the largest file, I guess, so maybe it's not any worse.